### PR TITLE
Implement Map.prototype[ @@toStringTag ]

### LINKED ISSUE #1229: Implement `Map.prototype[ @@toStringTag ]`
**ECMASCript feature**

Implement [`Map.prototype[ @@toStringTag ]`][spec], as a reference on how to implement it, you can check #1225 (`Math[ @@toStringTag ]`).

[spec]: https://tc39.es/ecma262/#sec-map.prototype-@@tostringtag

**Example code**
This code should now work and give the expected result:
```javascript
let result = (new Map())[Symbol.toStringTag];

console.log(result);
```
The expected output is `"Map"`.

### PR DESCRIPTION
**Pull request by wylie39**
This Implements `Map.prototype[ @@toStringTag ] `

This is based on #1225

resolves #1229

### CODE REVIEW

**Comment by Razican** [comment]
### Test262 conformance changes:
  | Test result | master count | PR count | difference |
  | :---------: | :----------: | :------: | :--------: |
  | Total | 78,701 | 78,701 | 0 |
  | Passed | 25,750 | 25,749 | **-1** |
  | Ignored | 15,519 | 15,519 | 0 |
  | Failed | 37,432 | 37,433 | **+1** |
  | Panics | 27 | 28 | **+1** |
  | Conformance | 32.72% | 32.72% | **-0.00%** |

Benchmarks:
```
┌─────────┬──────────────────────────────────────────────┬──────────────────────┬─────────────────────┬────────────┐
│ (index) │                     name                     │   changesDuration    │   masterDuration    │ difference │
├─────────┼──────────────────────────────────────────────┼──────────────────────┼─────────────────────┼────────────┤
│    0    │     'Arithmetic operations (Execution)'      │   '330.0±13.43ns'    │ '**324.1±14.65ns**' │   '+2.0'   │
│    1    │        'Arithmetic operations (Full)'        │   '228.8±11.36µs'    │ '**226.0±10.88µs**' │   '+1.0'   │
│    2    │          'Array access (Execution)'          │     '5.8±0.18µs'     │  '**5.4±0.23µs**'   │   '+6.0'   │
│    3    │            'Array access (Full)'             │   '256.2±12.01µs'    │ '**253.6±13.91µs**' │   '+1.0'   │
│    4    │         'Array creation (Execution)'         │     '2.9±0.10ms'     │  '**2.8±0.11ms**'   │   '+1.0'   │
│    5    │           'Array creation (Full)'            │     '3.0±0.12ms'     │  '**2.9±0.13ms**'   │   '+2.0'   │
│    6    │           'Array pop (Execution)'            │   '911.8±44.48µs'    │ '**901.7±41.36µs**' │   '+1.0'   │
│    7    │              'Array pop (Full)'              │ '**1281.8±46.70µs**' │  '1297.5±50.98µs'   │  '-0.99'   │
│    8    │     'Boolean Object Access (Execution)'      │   '**4.7±0.24µs**'   │    '4.8±0.22µs'     │  '-0.99'   │
│    9    │        'Boolean Object Access (Full)'        │ '**243.2±11.70µs**'  │   '252.8±9.80µs'    │   '-3.8'   │
│   10    │            'Clean js (Execution)'            │ '**604.8±26.59µs**'  │   '632.0±21.00µs'   │   '-3.8'   │
│   11    │              'Clean js (Full)'               │   '877.9±40.46µs'    │ '**869.6±33.34µs**' │   '+1.0'   │
│   12    │             'Clean js (Parser)'              │    '36.6±1.65µs'     │    '36.7±1.30µs'    │   '0.0'    │
│   13    │                'Create Realm'                │   '394.4±20.56ns'    │ '**389.2±15.14ns**' │   '+1.0'   │
│   14    │ 'Dynamic Object Property Access (Execution)' │     '4.6±0.19µs'     │  '**4.4±0.25µs**'   │   '+4.0'   │
│   15    │   'Dynamic Object Property Access (Full)'    │   '265.5±27.18µs'    │ '**256.2±13.05µs**' │   '+4.0'   │
│   16    │            'Expression (Parser)'             │     '6.2±0.31µs'     │  '**6.1±0.26µs**'   │   '+1.0'   │
│   17    │           'Fibonacci (Execution)'            │   '736.2±31.22µs'    │ '**687.8±24.11µs**' │   '+7.0'   │
│   18    │              'Fibonacci (Full)'              │   '949.0±45.71µs'    │ '**931.8±40.43µs**' │   '+2.0'   │
│   19    │            'For loop (Execution)'            │    '20.6±0.79µs'     │  '**19.9±0.96µs**'  │   '+3.0'   │
│   20    │              'For loop (Full)'               │   '268.0±14.12µs'    │ '**264.0±11.49µs**' │   '+2.0'   │
│   21    │             'For loop (Parser)'              │    '17.6±0.79µs'     │  '**17.5±0.82µs**'  │   '+1.0'   │
│   22    │           'Goal Symbols (Parser)'            │    '12.3±0.51µs'     │    '12.3±0.73µs'    │   '0.0'    │
│   23    │            'Hello World (Parser)'            │     '3.4±0.13µs'     │  '**3.3±0.15µs**'   │   '+4.0'   │
│   24    │             'Long file (Parser)'             │   '730.6±32.26ns'    │ '**715.2±32.56ns**' │   '+2.0'   │
│   25    │            'Mini js (Execution)'             │   '561.2±17.62µs'    │ '**529.3±19.66µs**' │   '+6.0'   │
│   26    │               'Mini js (Full)'               │ '**802.0±30.41µs**'  │   '811.4±36.55µs'   │  '-0.99'   │
│   27    │              'Mini js (Parser)'              │    '32.3±1.20µs'     │    '32.4±1.18µs'    │   '0.0'    │
│   28    │      'Number Object Access (Execution)'      │   '**3.6±0.17µs**'   │    '3.8±0.22µs'     │   '-3.8'   │
│   29    │        'Number Object Access (Full)'         │   '253.5±11.78µs'    │ '**249.1±10.44µs**' │   '+2.0'   │
│   30    │        'Object Creation (Execution)'         │   '**3.9±0.18µs**'   │    '4.0±0.10µs'     │   '-2.0'   │
│   31    │           'Object Creation (Full)'           │   '244.5±14.24µs'    │ '**242.4±12.57µs**' │   '+1.0'   │
│   32    │             'RegExp (Execution)'             │    '10.7±0.50µs'     │  '**10.5±0.57µs**'  │   '+2.0'   │
│   33    │               'RegExp (Full)'                │   '263.2±12.07µs'    │ '**255.1±15.00µs**' │   '+3.0'   │
│   34    │         'RegExp Literal (Execution)'         │  '**10.9±0.57µs**'   │    '11.3±0.40µs'    │   '-3.8'   │
│   35    │           'RegExp Literal (Full)'            │    '265.1±8.71µs'    │ '**254.4±12.04µs**' │   '+4.0'   │
│   36    │    'RegExp Literal Creation (Execution)'     │   '**9.2±0.48µs**'   │    '9.4±0.47µs'     │   '-2.9'   │
│   37    │       'RegExp Literal Creation (Full)'       │   '257.4±10.34µs'    │ '**250.4±12.04µs**' │   '+3.0'   │
│   38    │ 'Static Object Property Access (Execution)'  │     '4.2±0.18µs'     │  '**4.1±0.18µs**'   │   '+2.0'   │
│   39    │    'Static Object Property Access (Full)'    │ '**254.0±11.41µs**'  │   '259.7±9.67µs'    │   '-2.0'   │
│   40    │      'String Object Access (Execution)'      │     '6.5±0.25µs'     │    '6.5±0.30µs'     │   '0.0'    │
│   41    │        'String Object Access (Full)'         │   '251.9±11.09µs'    │   '251.6±11.60µs'   │   '0.0'    │
│   42    │       'String comparison (Execution)'        │     '5.9±0.34µs'     │    '5.9±0.27µs'     │   '0.0'    │
│   43    │          'String comparison (Full)'          │    '257.1±9.59µs'    │   '257.3±9.93µs'    │   '0.0'    │
│   44    │      'String concatenation (Execution)'      │     '4.9±0.22µs'     │  '**4.7±0.31µs**'   │   '+4.0'   │
│   45    │        'String concatenation (Full)'         │    '249.9±9.87µs'    │ '**242.4±11.01µs**' │   '+3.0'   │
│   46    │          'String copy (Execution)'           │     '3.5±0.18µs'     │  '**3.4±0.15µs**'   │   '+4.0'   │
│   47    │             'String copy (Full)'             │    '245.4±9.23µs'    │   '245.1±10.88µs'   │   '0.0'    │
│   48    │            'Symbols (Execution)'             │     '3.0±0.14µs'     │    '3.0±0.15µs'     │   '0.0'    │
│   49    │               'Symbols (Full)'               │    '234.3±8.82µs'    │ '**226.6±11.64µs**' │   '+3.0'   │
│   50    │                      ''                      │      undefined       │      undefined      │   '+NaN'   │
└─────────┴──────────────────────────────────────────────┴──────────────────────┴─────────────────────┴────────────┘
```

File: boa/src/builtins/map/mod.rs [hunk 1]
```diff
@@ -46,6 +48,11 @@ impl BuiltIn for Map {
                 entries_function.clone(),
                 Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE,
             )
+            .property(
+                to_string_tag,
+                "Math",
```
Comment (Razican) [inline boa/src/builtins/map/mod.rs:53]: Shouldn't this be `Map`?
```suggestion
                "Map",
```

**Review by Razican — CHANGES_REQUESTED** [review]

File: boa/src/builtins/map/mod.rs (hunk 1 above)
Comment (wylie39) [reply boa/src/builtins/map/mod.rs]: Yep, my bad.

**Review by Razican — APPROVED** [review]
Looks good from my side :) I updated the numbers, and now indeed it passes two extra tests.

**Review by HalidOdat — APPROVED** [review]

**Pull request merged**